### DISCUSSION TITLE
rabin2 error: Library handler has failed for 'core_pdd.so'

### DISCUSSION THREAD

**Discussion by ghost** [radareorg/radare2#22079] [General]
Hi! I'm not sure whether I should create an issue or not, but I caught a problem. I wrote a [StackOverflow post](https://stackoverflow.com/questions/75974688/rabin2-error-library-handler-has-failed-for-core-pdd-so) and happily forget about it. But a few days ago, it seems that someone caught it too. So, I guess, they need help.

I tried to run `rabin2` (`radare2` util) and got that error:

```bash
rabin2
# ERROR: Library handler has failed for '~/.local/share/radare2/plugins/core_pdd.so'
```

I've tried to google the error and got nothing: "Library handler has failed for" query with quotes and about 3 results, including `radare2` source code.

This error is definitely thrown from [here](https://github.com/radareorg/radare2/blob/4968d69f1848c2f9f2333d649a845009b59ba2a7/libr/util/lib.c#L365), but I can't figure out what exactly is wrong.

`radare2` was installed according to [author's instructions](https://github.com/radareorg/radare2#installation). [So was](https://github.com/wargio/r2dec-js#nix-users-linuxosxetc) `r2dec`. Also, I used `export PATH="$PATH:$HOME/.r2env/versions/radare2@git/bin"` to make it work.

At a quick glance, `pdd` and `r2dec` work pretty fine when executed from `r2` console.

Could you give me a clue how to fix that?

---

Versions:

```bash
r2 -v
# radare2 5.8.5 30144 @ linux-x86-64 git.5.8.4-98-g0cf6e7a5cc, 
# commit: 0cf6e7a5cc90554d735f4071b8b4888005e63428 build: 2023-04-08__22:52:24
uname -r
# 5.15.0-69-generic
lsb_release -a
# No LSB modules are available.
# Distributor ID: Ubuntu
# Description:    Ubuntu 22.04.2 LTS
# Release:        22.04
# Codename:       jammy
```

`r2dec` version: 5.7.8+ (commit [23819a6](https://github.com/wargio/r2dec-js/tree/23819a64c43d9f44fea94a14f308e343f3ab333f))

**Comment by trufae** [comment]
R2-5.8.5 is pretty old, please update. My assumption is that you have inpaired versions of libraries. You can run make purge in the r2 repo to remove all system installations of r2, so you can start clean from there

**Comment by trufae** [comment]
make purge
sys/install.sh
r2pm -ci r2dec
